opendatahub-io/odh-dashboard
# Trustyai demo phase0

Closes #1001 #1006 #1007 #1008

## Description


This completes the feature work for phase 0 of Explainability: Fairness and Bias. Features added:
- Tab based nav with deep linking added to modelServing views
- Charts for displaying both Disparate Inference Ratio and Statistical Parity Distance metrics
- Ability to switch between multiple payloads of the above metrics
- Tooltip displaying definitions of the above metrics

![odh-dashboard-trustyai-e2e apps acreasy wl4t s1 devshift org_projects_trustyai-e2e-modelmesh_metrics_model_demo-loan-nn-onnx-alpha_bias (2)](https://user-images.githubusercontent.com/4092230/233612494-990a636e-7e90-4958-8c37-50ed1114d42c.png)

![odh-dashboard-trustyai-e2e apps acreasy wl4t s1 devshift org_projects_trustyai-e2e-modelmesh_metrics_model_demo-loan-nn-onnx-alpha_bias (3)](https://user-images.githubusercontent.com/4092230/233612539-025b0435-e2bf-4ef4-af9e-fb476b635124.png)

## How Has This Been Tested?




### Test Environment Setup
* Deploy updated odh-dashboard to cluster
* Deploy main branch of TrustyAI in same namespace
* Through the dashboard, create a data science project, model server and model
* Send data to trustyAI service

### Testing
1. Open dashboard
2. Click 'Model Serving' in primary nav
3. In the table of available models, click the model name to open the metrics view
4. Click the bias tab
5. Verify SPD and DIR charts are displayed and plot data correctly
6. Verify the time dropdown and refresh buttons work correctly
7. Verify tooltips on both charts work correctly
8. Verify the payload selector on both charts functions correctly
9. Quickly repeat process on the second available model

## Test Impact


No tests added at this point, this is still a tech preview and subject to changes based on input from UX / TrustyAI. I think it makes more sense to look to add tests in later phases.

## Request review criteria:
